src/medida/stats/ckms_sample.h

```diff
@@ -0,0 +1,63 @@
+// Copyright 2021 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#ifndef MEDIDA_CKMS_SAMPLE_H_
+#define MEDIDA_CKMS_SAMPLE_H_
+
+#include <cstdint>
+#include <memory>
+
+#include "medida/types.h"
+#include "medida/stats/sample.h"
+#include "medida/stats/snapshot.h"
+
+namespace medida {
+namespace stats {
+
+// CKMSSample maintains two N-second windows: one for the current window, and
+// another for the previous window. It adds new data to the current one, and
+// it reports the previous one.
+//
+// For instance, if N = 30 and it's 1:00:45,
+// - it adds new data points to the current window [1:00:30, 1:01:00], and
+// - it reports the previous window [1:00:00, 1:00:30].
+
+
+// Each of size, Update, and MakeSnapshot has two versions, and
+// the one without a timestamp calls the other one with the current time.
+//
+// Unless there's a good reason to do so,
+// you should always use the one WITHOUT the timestamp.
+//
+// The one with a timestamp is generally used for testing. We pass a timestamp
+// as a way to fast-forward time to make testing easier.
+//
+// Regardless of which ones you use, the only rule that the caller must follow is
```

Comment:
This paragraph of this comment is no longer relevant, should be removed.

Reply:
I think it's still relevant. The class no longer throws when the caller goes back in time, but not going back in time is (I believe) the only simple way for the caller to not mess up. Going back in time by any amount, no matter how small, always has the risk of ending up in the wrong window.

Reply:
ok!